### PR TITLE
Des 185 dashborad page

### PR DESCRIPTION
**Pull request by Daniielmayo**
creation and completion of the page dashboard

### CODE REVIEW

**Review by abdulflorez — APPROVED** [review]

File: components/organisms/Login/Login.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
+"use client";
+import { FC, useEffect } from "react";
+import Link from "next/link";
+// import { useAuthStore } from ".././../../hooks";
```
Comment (DanielCalleCO) [inline components/organisms/Login/Login.tsx:4]: if you don't need this, delete it.

File: components/organisms/Login/Login.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+import { useAuthStore } from "../../../hooks/useAuthStore";
+import { useForm } from "../../../hooks/useForm";
+import { Button, Input } from "../../atoms";
+import styles from "./Login.module.css";
+import Swal from "sweetalert2";
+
+// interface LoginFormFields {
```
Comment (DanielCalleCO) [inline components/organisms/Login/Login.tsx:11]: delete or add an additional comment to explain why is there..

File: components/organisms/Login/Login.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+const loginFormFields: FormField = {
+  loginEmail: "",
+  loginPassword: "",
+};
+
+export const Login: FC = () => {
+  const { startLogin, errorMessage } = useAuthStore();
+
+  // const {
```
Comment (DanielCalleCO) [inline components/organisms/Login/Login.tsx:27]: remove

File: components/organisms/Login/Login.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    event.preventDefault();
+    startLogin({ email: loginEmail, password: loginPassword });
+    console.log("echo en logincomponent");
+  };
+
+  useEffect(() => {
+    if (errorMessage !== undefined)
+      Swal.fire("Usuario o contraseña incorrecta", errorMessage, "warning");
+  }, [errorMessage]);
+  // const {
```
Comment (DanielCalleCO) [inline components/organisms/Login/Login.tsx:46]: remove

File: hooks/useAuthStore.ts [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  email: string;
+  country: string;
+  city: string;
+  phone: string;
+  uid: string;
+  token: string;
+  image: string;
+}
+
+// export interface AuthStore {
```
Comment (DanielCalleCO) [inline hooks/useAuthStore.ts:23]: remove

File: middlewares/middlewareAuh.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+
+// This function can be marked `async` if using `await` inside
+export async function middleware(req: NextRequest) {
+  const sesionAuth = await getToken({
+    req,
+    // secret: process.env.NEXT_PUBLIC_API_URL,
+  
+  });
+  console.log(sesionAuth);
```
Comment (DanielCalleCO) [inline middlewares/middlewareAuh.ts:13]: remove this clg and the next comment... or add a comment explaining.

File: store/auth/authSlice.tsx [hunk 7]
```diff
@@ -0,0 +1,95 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+
+// Define el tipo para el estado inicial
```
Comment (DanielCalleCO) [inline store/auth/authSlice.tsx:3]: translation

File: store/auth/authSlice.tsx [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+export interface AuthState {
+  status: string;
+  user: { uid?: string; email?: string };
+  errorMessage?: string | undefined;
+}
+
+// Define el tipo para las acciones
```
Comment (DanielCalleCO) [inline store/auth/authSlice.tsx:10]: translation

File: components/organisms/DashboardUserWelcome/DashboardUserWelcome..tsx [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+          ¡Continúa tu curso!
+        </h2>
+        <img
+          className={styles["continue-course_video"]}
+          src="https://static.wixstatic.com/media/d166cc_4cc837baf9254000a0f3963193c6b07a~mv2.jpg/v1/fill/w_368,h_195,al_c,q_80,usm_0.66_1.00_0.01,enc_auto/Romanos%2011111.jpg"
+          alt=""
+        />
+      </div>
+      <div className={styles["dashboard-continue_course"]}>
+        {/* <h2 className={styles["continue-course_subtitle"]}>¡Recursos para ti!</h2> */}
```
Comment (DanielCalleCO) [inline components/organisms/DashboardUserWelcome/DashboardUserWelcome..tsx:45]: do we need this?

**Review by DanielCalleCO — CHANGES_REQUESTED** [review]

File: components/organisms/DashboardUserWelcome/DashboardUserWelcome..tsx (hunk 9 above)
Comment (DanielCalleCO) [reply components/organisms/DashboardUserWelcome/DashboardUserWelcome..tsx:45]: once you solve this, re-request my review please

File: hooks/useAuthStore.ts [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+  uid: string;
+  token: string;
+  image: string;
+}
+
+
+interface AuthStore {
+  status: string;
+  errorMessage: string | undefined;
+  user: Partial<UserData>; // Permite propiedades opcionales en user
```
Comment (DanielCalleCO) [inline hooks/useAuthStore.ts:27]: translate

File: hooks/useAuthStore.ts [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+      localStorage.setItem("city", data.city);
+      localStorage.setItem("phone", data.phone);
+      localStorage.setItem("id", data.uid);
+      localStorage.setItem("token", data.token);
+      localStorage.setItem("image", data.image);
+      localStorage.setItem("token-init-date", String(new Date().getTime()));
+
+     dispatch(onLogin({ data, token: data.token }));
+        console.log(data);
+      console.log("autenticado");
```
Comment (DanielCalleCO) [inline hooks/useAuthStore.ts:69]: remove

**Review by DanielCalleCO — CHANGES_REQUESTED** [review]

**Review by DanielCalleCO — APPROVED** [review]

**Pull request merged**